internal/controller/datadogagentinternal/controller_reconcile_v2.go

```diff
@@ -64,19 +65,22 @@ func (r *Reconciler) reconcileInstanceV2(ctx context.Context, logger logr.Logger
 	r.updateMetricsForwardersFeatures(instanceCopy, enabledFeatures)
 
 	// 1. Manage dependencies.
-	depsStore, resourceManagers := r.setupDependencies(instanceCopy, logger)
+	// set the original DDAI as the owner of dependencies
+	depsStore, resourceManagers := r.setupDependencies(instance, logger)
```

Comment:
We're using `instance` again instead of `instanceCopy` -> the default DDAI instance uses the same name as DDA, should we remove the lines 60 and 61 ?